### PR TITLE
[Promotion][Admin][Core] Actions with amounts per channel

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets | |
| License         | MIT |

To-do:
- [x] small fixes in failing build
- [x] fix UI as it looks terrible 😕

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/BuildChannelAwarePromotionActionFormSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ */
+final class BuildChannelAwarePromotionActionFormSubscriber extends BuildPromotionActionFormSubscriber
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @inheritDoc
```
Comment (tuka217) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/BuildChannelAwarePromotionActionFormSubscriber.php]: `{@inheritdoc}` or shouldn't it be comment with names of parameters?

File: src/Sylius/Bundle/CoreBundle/Form/Type/Promotion/PromotionActionType.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+namespace Sylius\Bundle\CoreBundle\Form\Type\Promotion;
+
+use Sylius\Bundle\CoreBundle\Form\EventSubscriber\BuildChannelAwarePromotionActionFormSubscriber;
+use Sylius\Bundle\PromotionBundle\Form\Type\Core\AbstractConfigurationType;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Registry\ServiceRegistryInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * {@inheritdoc}
```
Comment (tuka217) [inline src/Sylius/Bundle/CoreBundle/Form/Type/Promotion/PromotionActionType.php]: ?

File: features/promotion/receiving_discount/receiving_fixed_discount_dependent_on_channel.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@receiving_discount
+Feature: Receiving fixed discount dependent on channel on cart
+    In order to pay proper amount while buying promoted goods in different channels
+    As a Visitor
+    I want to have promotions applied to my cart
+
+    Background:
+        And the store operates on a channel named "Web-US" in "USD" currency
```
Comment (GSadee) [inline features/promotion/receiving_discount/receiving_fixed_discount_dependent_on_channel.feature]: `Given`

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/BuildChannelAwarePromotionActionFormSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class BuildChannelAwarePromotionActionFormSubscriber extends BuildPromotionActionFormSubscriber
+{
+    /**
+     * @var ChannelRepositoryInterface
+     */
+    private $channelRepository;
+
+
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/BuildChannelAwarePromotionActionFormSubscriber.php]: Too many lines

File: features/promotion/managing_promotions/adding_promotion_with_action_in_different_channels.feature [hunk 5]
```diff
@@ -1,24 +1,21 @@
...
-    I want to add a new promotion with action configured in different currencies to the registry
+    I want to add a new promotion with action configured in different channels to the registry
 
     Background:
-        Given the store has currency "USD"
-        And the store has currency "GBP"
-        And the store operates on a channel named "Web-US" in "USD" currency
-        And that channel allows to shop using "USD" and "GBP" currencies
+        Given the store operates on a single channel in "United States"
+        And the store operates on a channel named "Web-GB" in "GBP" currency
```
Comment (michalmarcinkowski) [inline features/promotion/managing_promotions/adding_promotion_with_action_in_different_channels.feature]: the store **also** operates

Comment (michalmarcinkowski) [inline features/promotion/managing_promotions/adding_promotion_with_action_in_different_channels.feature]: `And the store also operates on another channel named ...` - we can reuse this step

File: src/Sylius/Component/Core/Factory/PromotionActionFactoryInterface.php [hunk 6]
```diff
@@ -20,37 +20,42 @@
...
+    public function createUnitPercentageDiscount($percentage, $channelCode);
 
     /**
      * @param float $percentage
+     * @param string $channelCode
      *
      * @return PromotionActionInterface
      */
-    public function createShippingPercentageDiscount($percentage);
+    public function createShippingPercentageDiscount($percentage, $channelCode);
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Factory/PromotionActionFactoryInterface.php]: Percentage discounts should not be channel aware.

File: src/Sylius/Component/Core/Promotion/Action/FixedDiscountPromotionActionCommand.php [hunk 7]
```diff
@@ -59,12 +59,18 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
             return;
         }
 
-        $this->isConfigurationValid($configuration);
+        $channelCode = $subject->getChannel()->getCode();
+        if (!isset($configuration[$channelCode])) {
+            return;
```
Comment (Zales0123) [inline src/Sylius/Component/Core/Promotion/Action/FixedDiscountPromotionActionCommand.php]: I think we should not - if there is no configuration for channel it simple means there should be no discount on that channel. It's not necessarily an error.

**Comment by Zales0123** [comment]
Closing in favor of https://github.com/Sylius/Sylius/pull/6906

**Pull request closed without merging**